File: javascript/p0001.js
/**
* Project Euler Problem 1
* 
* Did this the old fashioned way, because this was before I figured out the closed form solution
* 
* Problem:
* 
* If we list all the natural numbers below 10 that are multiples of 3 or 5, we
* get 3, 5, 6 and 9. The sum of these multiples is 23.
* 
* Find the sum of all the multiples of 3 or 5 below 1000.
*/
exports.p0001 = function() {
    let answer = 0;
    for (let i = 3; i < 1000; i += 3) {
        answer += i;
    }
    for (let i = 5; i < 1000; i += 5) {
        answer += i;
    }
    for (let i = 15; i < 1000; i += 15) {
        answer -= i;
    }
    return answer;
};

File: rust/src/p0001.rs
/*
Project Euler Problem 1

I did this in the traditional way, as I'm mostly using this folder as a way
to learn Rust

Problem:

If we list all the natural numbers below 10 that are multiples of 3 or 5, we
get 3, 5, 6 and 9. The sum of these multiples is 23.

Find the sum of all the multiples of 3 or 5 below 1000.
*/
pub fn p0001() -> i128 {
    let mut answer: u32 = 0;
    for i in (0..1000).step_by(3) {
        answer += i;
    }
    for i in (0..1000).step_by(5) {
        answer += i;
    }
    for i in (0..1000).step_by(15) {
        answer -= i;
    }
    return answer.into();
}
